src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -56,8 +78,34 @@ static ucs_status_t uct_ib_mlx5_dereg_key(uct_ib_md_t *md,
                                           uct_ib_mr_type_t mr_type)
 {
     uct_ib_mlx5_mem_t *memh = ucs_derived_of(ib_memh, uct_ib_mlx5_mem_t);
+    int ret;
 
-    return uct_ib_dereg_mr(memh->mrs[mr_type].super.ib);
+    if (memh->type == UCT_IB_MLX5_MEM_REG) {
+        return uct_ib_dereg_mr(memh->mrs[mr_type].super.ib);
+    }
+
+    ucs_assert_always((memh->type == UCT_IB_MLX5_MEM_EXPORTED) ||
+                      (memh->type == UCT_IB_MLX5_MEM_IMPORTED));
+
+    if (mr_type != UCT_IB_MR_DEFAULT) {
```

Comment:
is it possible for non `UCT_IB_MLX5_MEM_REG` key?

Reply:
not possible added `assert`